pkg/api/v1alpha1/nutanixmachineconfig_webhook.go

```diff
@@ -0,0 +1,144 @@
+			in.Name,
+			field.ErrorList{
+				field.Invalid(field.NewPath("spec"), in.Spec, err.Error()),
+			},
+		)
+	}
+
+	return nil
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type.
+func (in *NutanixMachineConfig) ValidateUpdate(old runtime.Object) error {
+	nutanixmachineconfiglog.Info("validate update", "name", in.Name)
+
+	oldNutanixMachineConfig, ok := old.(*NutanixMachineConfig)
+	if !ok {
+		return apierrors.NewBadRequest(fmt.Sprintf("expected a NutanixMachineConfig but got a %T", old))
+	}
+
+	var allErrs field.ErrorList
+	if err := in.Validate(); err != nil {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec"), in.Spec, err.Error()))
+	}
+
+	if oldNutanixMachineConfig.IsReconcilePaused() {
+		nutanixmachineconfiglog.Info("NutanixMachineConfig is paused, so allowing create", "name", in.Name)
+		if len(allErrs) > 0 {
+			return apierrors.NewInvalid(
+				GroupVersion.WithKind(NutanixMachineConfigKind).GroupKind(),
+				in.Name,
+				allErrs,
+			)
+		}
+		return nil
+	}
+
+	allErrs = append(allErrs, validateImmutableFieldsNutantixMachineConfig(in, oldNutanixMachineConfig)...)
+	if len(allErrs) > 0 {
+		return apierrors.NewInvalid(
+			GroupVersion.WithKind(NutanixMachineConfigKind).GroupKind(),
+			in.Name,
+			allErrs,
+		)
+	}
+
+	return nil
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type.
+func (in *NutanixMachineConfig) ValidateDelete() error {
+	nutanixmachineconfiglog.Info("validate delete", "name", in.Name)
+
+	// TODO(user): fill in your validation logic upon object deletion.
+	return nil
+}
+
+func validateImmutableFieldsNutantixMachineConfig(new, old *NutanixMachineConfig) field.ErrorList {
+	var allErrs field.ErrorList
+	specPath := field.NewPath("spec")
+	if new.Spec.OSFamily != old.Spec.OSFamily {
+		allErrs = append(allErrs, field.Forbidden(specPath.Child("OSFamily"), "field is immutable"))
+	}
+
+	if !reflect.DeepEqual(new.Spec.Cluster, old.Spec.Cluster) {
+		allErrs = append(allErrs, field.Forbidden(specPath.Child("Cluster"), "field is immutable"))
+	}
+
+	if !reflect.DeepEqual(new.Spec.Subnet, old.Spec.Subnet) {
+		allErrs = append(allErrs, field.Forbidden(specPath.Child("Subnet"), "field is immutable"))
+	}
+
+	if old.IsManaged() {
+		nutanixmachineconfiglog.Info("Machine config is associated with workload cluster", "name", old.Name)
+		return allErrs
+	}
+
+	if !old.IsEtcd() && !old.IsControlPlane() {
+		nutanixmachineconfiglog.Info("Machine config is associated with management cluster's worker nodes", "name", old.Name)
+		return allErrs
+	}
+
+	nutanixmachineconfiglog.Info("Machine config is associated with management cluster's control plane or etcd", "name", old.Name)
+
+	if err := validateImmutableFieldsControlPlane(new, old); err != nil {
+		allErrs = append(allErrs, err...)
+	}
+
+	return allErrs
+}
+
+func validateImmutableFieldsControlPlane(new, old *NutanixMachineConfig) field.ErrorList {
```

Comment:
Here, I think we need this for control plane upgrades to work through the EKS-A CLI.
https://github.com/aws/eks-anywhere/blob/b01fd2e046ad6463b6cf4e83d565cf2bae39133a/pkg/api/v1alpha1/vspheremachineconfig_webhook.go#L59-L62 
The CLI adds the pause annotation to when performing CLI operations which would allow updating the management cluster's control planes. Once the CLI is done with the upgrade, it removes this pause annotation so the controller doesn't try to modify the management cluster's control plane